fix issue #511

mrtnmgs opened the pull request:
Fix issue #511 Visiting the site with javascript disabled does not display any projects.
Only the list displays (no filtering by tag). Also, the label doesn't display how many issues have this label (as the JS page does).
This is my first PR! Sorry if I'm doing anything wrong!

aph3rson commented on index.html:
```diff
@@ -8,8 +8,39 @@
 <div class="block">
   <h1><span class="header-inside">Projects</span></h1>
   <div id="projects-panel">
+    <table class="projects">
```
Your `<table>` tags are outside of your `<noscript>` tags. As far as I can tell, this will cause issues - you should check to see if there's an empty set of `<table>`s if rendering the site with JS turned on.

mrtnmgs replied:
Good catch thanks!

aph3rson commented on index.html:
```diff
@@ -8,8 +8,39 @@
+          </tr>
+          <tr>
+            <td class="details">
+              <p class="label"><a href="{{ project.upforgrabs.link }}" title="View open issues for {{ project.name }}">{{ project.upforgrabs.name }}</a></p>
```
You mention open issues - however, it's guaranteed that that link will take you to a list of issues for that project. Some have added links to their contributors' guidelines, and some to a discussion board (iirc).

It might be better to generalize this verbiage, to something like "Contribute to XYZ project!"

mrtnmgs replied:
I reused the language from the js page

aph3rson commented on index.html:
```diff
@@ -8,8 +8,39 @@
+              {% if project.tags %}
+                <p class="tags">
+                  {% for tag in project.tags %}
+                    {{ tag }}{% if forloop.index != forloop.length %},{% endif %}
```
There should be a space after that comma, else you're going to see a list like this:  
tag1,tag2,tag3

For future reference, you can simplify that if statement like this:

```html
{{ tag }}{% if not forloop.last %}, {% endif %}
```

mrtnmgs replied:
Thanks!

aph3rson commented on index.html:
```diff
@@ -8,8 +8,39 @@
+    <table class="projects">
+      <noscript>
+        {% for p in site.data.projects %}
+        {% assign project = p[1] %}
```
Just to check, do you need that array index? IIRC, `site.data.projects` is already an array of projects, and your forloop should be going through each project iteratively - what you've got looks like it's an array of arrays.

mrtnmgs replied:
This is how `site.data.projects` is structured – it doesn't work without line 14. There might be more elegant ways to iterate but I didn't find... I'm not really sure why the data is structured like this in the first place.

mrtnmgs commented on index.html:
```diff
@@ -8,8 +8,39 @@
+              {% if project.tags %}
+                <p class="tags">
+                  {% for tag in project.tags %}
+                    {{ tag }}{% if forloop.index != forloop.length %},{% endif %}
```
@aph3rson Do I have to do anything to add your changes to my commit? This is my first PR I feel a bit in the dark!

aph3rson replied:
You'll just need to commit to the branch on your side of the PR. The PR will update automatically, and your commits will be shown here.

nicolas-raoul committed:
Added Wikivoyage listings

mjgpy3 committed:
Add FSharp.Data.JsonValidation

Commit:
Create room-sprawl.yml

Commit:
Update room-sprawl.yml

fixed list filtering

ifeanyidavid committed:
#682 - Added functionality to allow external project links open in a separate tab

mfkl committed:
add vlc-winrt project

mfkl committed:
removed comment

Commit:
Add ApplicationInsights-dotnet

Commit:
remove extra newlines

Commit:
quotes for .NET tag

sobolevn committed:
Adds git-secret

`git-secret` is looking for additional maintainers: https://github.com/sobolevn/git-secret/issues/100

larrysalibra committed:
add blockstack browser

larrysalibra committed:
add blockstack.js. fix typo

EdwardDunn committed:
Added project file Platform_Game.yml

elijahcruz12 committed:
Create eckit.yml

Griffintaur committed:
Adding MediaPlayer Project

EdwardDunn committed:
Changed label name to up-for-grabs

hazzik committed:
Create DelegateDecompiler.yml

hazzik committed:
Update DelegateDecompiler.yml

- quote .NET tag.
- add bunch of other tags

bigpresh committed:
Add dancer2.yml

jskeet committed:
Add Noda Time project data

Commit:
Added LocalAppVeyor project

Commit:
made tags to be between quotes

Commit:
added csharp tag

jskeet committed:
Add the .net tag for Noda Time

Commit:
Add aptos project

Commit:
Add 'python' to tags

pierredup committed:
Add SolidInvoice

sallakarppinen committed:
Created a file for OAE Project Hilary (back-end)

OAE Project is an open source project aiming to create a new digital environment for academia.

StanczakDominik committed:
Add PlasmaPy project

blavejr committed:
Create OrganiseDesktop.yml

blavejr committed:
Update OrganiseDesktop.yml

blavejr committed:
Update OrganiseDesktop.yml

blavejr committed:
proper link and issues labeled help wanted

horazont committed:
Add aioxmpp

akosyakov committed:
Create Theia.yml

akosyakov committed:
Update the link

Signed-off-by: Anton Kosiakov <anton.kosyakov@typefox.io>

akosyakov committed:
Added Monaco language client project

Signed-off-by: Anton Kosiakov <anton.kosyakov@typefox.io>

Motardo committed:
Add filter by name

guusdk committed:
Create Openfire.yml

guusdk committed:
Create Spark.yml

brylie committed:
Add CiviWiki

chfast committed:
Add Hunter project

chfast committed:
Hunter: rename label to "help wanted"

mxstbr committed:
Add the styled-components website

Adds https://github.com/styled-components/styled-components-website

ritwik12 committed:
Add Virtual-Assistant.yml

jbapple-cloudera committed:
Add Apache Impala

bricelam committed:
Add Microsoft.Data.Sqlite

ForNeVeR committed:
Avalonia: rename from Perspex

jacobwardio committed:
Create hugo-cactus-theme-2017.yml

zkochan committed:
Add pnpm project

ChrisMissal committed:
Create odota-core.yml

ChrisMissal committed:
Create odota-ui.yml

ChrisMissal committed:
Create odota-mobile.yml

jacobtomlinson committed:
Add opsdroid

nkmlombardi committed:
Create blockstreet.yml

layely committed:
Fix issue #511 Visiting site without javascript

jacobtomlinson committed:
Rename file

nkmlombardi committed:
Update blockstreet.yml

rohithvootla committed:
Added "site-www" to the projects directory

Added a new project for beginners as requested at dart-lang/site-www#341.

nkmlombardi committed:
Update blockstreet.yml

zachkont committed:
Create dotaUpdatesBot project file

DreadKnight committed:
added Ancient Beast game project

looking forward to getting new coders involved

stoyicker committed:
Add stoyicker/dinger

mayukh18 committed:
Add blindchat

a messenger app built on flask

andy5995 committed:
add project:if me

Hi @shiftkey
I have permission from the project manager to add this.
If you need to confirm, you may contact @julianguyen

AlekSi committed:
Add reform.

AlekSi committed:
Use lowercase «orm» everywhere.

AlekSi committed:
Use lowercase «go» everywhere.

It’s not «golang».

DreadKnight committed:
Improved Ancient Beast description

Nicely fits into the website's paragraph.

andy5995 committed:
ifme:entry label changed

mayukh18 committed:
Corrected file format

Corrected to .yml format

smarter committed:
Add Dotty, the next-generation Scala compiler

Pamblam committed:
created jquery plugin

Maxwell175 committed:
Create SSNFS.yml

shubha1593 committed:
Create Oppia Project

andy5995 committed:
after.html:changed _tag_ to _label_

_Label_ is much more specific

andy5995 committed:
tag -> label

andy5995 committed:
mhwkb.yml:added new project

Maxwell175 committed:
Fix tags to be in line with current ones.

basedalexander committed:
Create container-ioc.yml

mubaris committed:
Add motivate

WaveHack committed:
Add OpenDominion project

andy5995 committed:
updating tags

szymczakk committed:
remove makesharp - no longer maintained

jaguililla committed:
Add Hexagon to the projects list

* https://github.com/hexagonkt/hexagon
* https://github.com/hexagonkt/hexagon/projects/1

Croydon committed:
Update inexor.yml

URL to repository changed due to renaming, label changed due to GitHub's new use of "good first issue".

koppor committed:
Rename "beginner" to "good first issue"

uralbash committed:
Create lecturesWWW.yml

uralbash committed:
Update lecturesWWW.yml

uralbash committed:
Update lecturesWWW.yml

phw committed:
Peek now uses "good first issue" tag

WaveHack committed:
Change OpenDominion label to 'good first issue'

ghuntley committed:
help wanted

ghuntley committed:
help wanted

ghuntley committed:
:ship:

ghuntley committed:
Update reactiveui-website.yml

ghuntley committed:
Update reactiveui-website.yml

Commit:
Use "help wanted" label for NativeScript

We have renamed the "up-for-grabs" label to "help wanted"

RishabhJain2018 committed:
Add EvalAI project with description.

Commit:
Search link changed for the whole NS org

BillWagner committed:
Update allready.yml

We're moving to the standard github labels for Help Wanted and Good First issue.

Maxwell175 committed:
Update SSNFS.yml

shubha1593 committed:
Update label for Oppia.yml

kr-stn committed:
new project: sentinelsat

jchannon committed:
Add Botwin project

dilijev committed:
ChakraCore: update to use label "help wanted"

dilijev committed:
ChakraCore: Update tags

Hrabal committed:
Create tempy.yml

prabhakar267 committed:
Add Library Management System

Bond-009 committed:
Create cuberite

comodoro committed:
Add FractalZoo project

miparnisari committed:
Create node-cognitive-services.yml

RamblingCookieMonster committed:
Add PSSlack project

Maxwell175 committed:
Fix Link

jchannon committed:
Update Botwin.yml

suriyaa committed:
Add "Animate A Name"

Kissaki committed:
Add Mumble project

lgoetz committed:
Add the Shogun Machine Learning Toolbox project

woodruffw committed:
Create KBSecret.yml

christianp committed:
Numbas uses "good first issue" label now

GitHub recommended it.

christianp committed:
Numbas editor now uses "good first issue" label

As recommended by GitHub

ttu committed:
Add Fake JSON Server

ttu committed:
Add JsonFlatFileDataStore

lgoetz committed:
Rename shogun to shogun.yml

alexweissman committed:
Add UserFrosting.yml

 UserFrosting is a web framework for PHP, built with widely-used components like Laravel's Eloquent ORM, Slim PHP, and Twig. Like other web frameworks, it offers a complete toolbox of programmatic components for building a modern web application.  Unlike other PHP frameworks it's a fully-functioning user login and management application, right out of the box.

andy5995 committed:
mhwkb: update link, add tag, change label

bgub committed:
Create squirrelly.yml

For the squirrelly project

carlspring committed:
Added the strongbox project.

carlspring committed:
Added the strongbox project.

andy5995 committed:
mhwkb:fix name

ChristianAlexander committed:
Update influxdb help wanted label.

DanTup committed:
Switch to "good first issue" since GH apparently treats it special!

carlspring committed:
Added the strongbox project.

tiltec committed:
Change foodsaving tool to karrot, update description

ColinMoulds committed:
Create Swat-OS.yml

ColinMoulds committed:
Update Swat-OS.yml

ColinMoulds committed:
Update Swat-OS.yml

ColinMoulds committed:
Update Swat-OS.yml

andy5995 committed:
mhwkb: update project info

zvonimirr committed:
added wikirider

zvonimirr committed:
fix yml

StanczakDominik committed:
Create PlasmaPyWebsite.yml

Hi!

UpForGrabs has been awesome for our project's visibility. We're working on a website though, and we could use some help with that as well as we're completely in the dark about web development, basically. Do you think work with you guys on this one as well?

Thanks for everything!

SunyataZero committed:
Create mindfulness-at-the-computer.yml

TheMushrr00m committed:
Adding cordovue project

chfast committed:
Update cpp-ethereum.yml

Commit:
Create ancestorquotes.yml

jarun committed:
Add ddgr.yml.

ddgr is a new project I am working on. It enables one to search DuckDuckGo from
the terminal.

I noticed a high demand for this utility in several discussions involving
googler, my other project to google from the terminal. Privacy is of the topmost
priority for many internet users and DuckDuckGo does a fine job at that.

It would be great to have some collaborators to work with and discuss. Some of
my utilities are already up-for-grabs. Can't think of a better place!

ritwik12 committed:
Update Virtual assistant.yml

change label from "jump in" to "good first issue"

TheMushrr00m committed:
Update cordovue.yml

sapegin committed:
Update React Styleguidist label

jernst committed:
Updated starter issue github tag per their recent convention change

andy5995 committed:
htdig: add project

gambry committed:
Update link to Drupal Novice issues queue

Commit:
Popcorn: add Popcorn project .yml

Commit:
Popcorn: Fix naming of tag

Commit:
Fixing issue with ".NET" tag

ganderzz committed:
Create Gnerate.yml

Add gnerate to the list

jaguililla committed:
Update hexagon.yml

codybartfast committed:
Add xUnit.net project file

bradwilson committed:
Put .NET tag in quotes (per CI)

andy5995 committed:
c4sg-web: add project

andy5995 committed:
c4sgweb.yml: rename yml file, correct grammar in desc

andy5995 committed:
c4sgweb.yml: change link to repo

Fixed link

jaguililla committed:
Update hexagon.yml

zvirja committed:
Update AutoFixture tag info

metaprime committed:
ripme.yml: ripme has moved to RipMeApp/ripme

hazzik committed:
Add NHibernate

shrayasr committed:
Add Iko

shrayasr committed:
Add TheHexBot

abhijitnathwani committed:
Create PyQuiz.yml

rofr committed:
Create memstate.yml

Adding memstate, a high-performance in-memory event-sourced replicated object graph engine for dotnet

rofr committed:
Update memstate.yml

changed url to point at github repo

jabdoa2 committed:
change label according to github specs

Griffintaur committed:
adding Padestrian-Detection project

BlythMeister committed:
Create servicebouncer.yml

BlythMeister committed:
Correct typo in service bouncer yml

chfast committed:
Update ethminer: [up-for-grubs] -> [help wanted]

axic committed:
Add Solidity

Simply007 committed:
added .net delivery sdk

Simply007 committed:
added sample React appliation

Simply007 committed:
added java delivery sdk

Simply007 committed:
added .net bolerplate

Simply007 committed:
added accelerated mobile pages module

Simply007 committed:
added .net model generator

Simply007 committed:
added -net's tag in quotes

Commit:
Create CPlot.yml

Commit:
Update CPlot.yml

Space --> %20

phenax committed:
Added phenax/h

phenax committed:
Update H.yml

andy5995 committed:
rmw:add project

andy5995 committed:
rmw:link correction

tdwright committed:
Adding ConTabs to the up-for-grabs roster

tdwright committed:
Fixed indentation of link property

phenax committed:
Changed link to all help-wanted issues

metaprime committed:
Update ripme.yml to use "help wanted" tag (See RipMeApp/ripme#298)

https://github.com/RipMeApp/ripme/issues/298

faraazahmad committed:
Update shellrb.yml

Findus23 committed:
add Piwik

Findus23 committed:
replace "website" with "web"

mairaw committed:
update dotnet/docs info

bruno-garcia committed:
Create greentube.serialization.yml

 Greentube.Serialization's up for grabs

bruno-garcia committed:
Use the more common tag ".NET" and C#

andy5995 committed:
htdig: update tag

andy5995 committed:
mhwkb: change title

andy5995 committed:
rmw: update tags and description

zvonimirr committed:
Added passthief

treyhboy committed:
Add Mint.yml

added my project Mint

treyhboy committed:
Update Mint.yml

treyhboy committed:
Update Mint.yml

andy5995 committed:
hldig: rename from htdig

Commit:
Not quite good for newbies anymore...

Too much major work going on, and I'm the only contributor

lwindolf committed:
Add Liferea (Linux GTK/GNOME feed reader)

andy5995 committed:
hldig: add tags

marek-safar committed:
Add Mono to project list

lwindolf committed:
Fix indentation and link

jedeen committed:
Update tag

phenax committed:
Added pattern-lock-js

dabutvin committed:
updates dotnet/roslyn from up-for-grabs to help+wanted

faheel committed:
Add BigInt

brockallen committed:
Create IdentityServer4.yml

ilexp committed:
Updated duality.yml With New Label Name

Renamed the Up for Grabs label in the project, updating the .yml file to match.

SunyataZero committed:
Updated the github project page

after having moved the project into an organization

Commit:
Made a typo. (+1 squashed commit)
Squashed commits:
[9382b51] Adding UnityStation to up-for-grabs

brockallen committed:
Update IdentityServer4.yml

ritwik12 committed:
Revert "Pagination (#665) (#303)"

dxrsz committed:
Fix searching for project names with whitespace not returning results.

swapagarwal committed:
Create medium-play.yml

dhruvapte26 committed:
Added first-timers-guide

dhruvapte26 committed:
Corrected to allow page build

Commit:
Create GoGE.yml

Commit:
Add upforgrabs

Commit:
Update GoGE.yml

andy5995 committed:
zegaglest.yml: add project

gfiedler committed:
Create openstreetcraft.yaml

gfiedler committed:
Update openstreetcraft.yaml

gfiedler committed:
Rename openstreetcraft.yaml to openstreetcraft.yml

shiftkey committed:
bump lockfile to address nokogiri security issue

andy5995 committed:
update description

evanplaice committed:
Create jquery-csv.yml

Laying the foundation to make jquery-csv contributor-friendly.

the-wendell committed:
Create my-dev-portal.yml

This is my first time setting up a project for open source so please let me know if I have overlooked anything.

Cheers,
Scott

p.s. Thank you for building and maintaining up-for-grabs!

feat7 committed:
Create Surface.yml

bertocq committed:
Update consul's label name

We've changed the label from `beginner-friendly` on to `Good first issue`

andy5995 committed:
hldig:remove project

jancborchardt committed:
Change "starter issue" to "good first issue" for Nextcloud

andy5995 committed:
rmw: remove project

andy5995 committed:
ZetaGlest:update info

* added some tags
* changed the link to the issues so that tickets from multiple repos
from within the ZetaGlest organization will be displayed.

Commit:
create mentorient.yml

add a new project

faraazahmad committed:
made changes to the link

DecentM committed:
Add OpenUserCSS project yml

BeastCode committed:
Create VSCode-Angular-TypeScript-Snippets.yml

Up for grabs issues in https://github.com/BeastCode/VSCode-Angular-TypeScript-Snippets
